misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -322,6 +334,22 @@ private Document getDocumentFromModule(Module moduleInstance, String sourceFileN
         return exclusionPathList;
     }
 
+    private List<Path> getDependencyJarList(JBallerinaBackend jBallerinaBackend) {
```

Comment:
Check getJarFilePathsRequiredForExecution in JarResolver I guess you can get the dependencies list directly from that

Reply:
We still need to remove the runtime libraries and test dependencies to avoid errors as discussed.